Engine/src/Engine/UserInterface.cpp

```diff
@@ -0,0 +1,101 @@
+#include "UserInterface.h"
+
+ImGuiIO* UserInterface::s_IO = nullptr;
+ImGuiStyle* UserInterface::s_style = nullptr;
+
+std::map<std::string, ImFont*> UserInterface::s_fonts;
+
+//define the ImGUI vec4s for each colour
+ImVec4 UserInterface::red = ImVec4(1.0f, 0.0f, 0.0f, 1.0f);
+ImVec4 UserInterface::green = ImVec4(0.0f, 1.0f, 0.0f, 1.0f);
+ImVec4 UserInterface::blue = ImVec4(0.0f, 0.0f, 1.0f, 1.0f);
+ImVec4 UserInterface::white = ImVec4(1.0f, 1.0f, 1.0f, 1.0f);
+
+//initializes a GLFWwindow that can be used with ImGUI
+bool UserInterface::initialize(GLFWwindow* window)
+{
+	ImGui::CreateContext();
+	ImGui_ImplOpenGL3_Init("#version 330");
+	ImGui_ImplGlfw_InitForOpenGL(window, true);
+
+	//TODO - Reference for style
+	//https://www.unknowncheats.me/forum/c-and-c-/189635-imgui-style-settings.html
+	s_style = &ImGui::GetStyle();
+
+	s_IO = &ImGui::GetIO();
```

Comment:
What's IO?

Reply:
ImGui IO is used later for loading custom fonts.

Reply:
can you mention that wherever s_IO is defined via comment?

Reply:
@KevinAlexV added comment